server/libs/domain/src/job/job.service.ts

```diff
@@ -110,6 +110,10 @@ export class JobService {
         await this.jobRepository.queue({ name: JobName.STORAGE_TEMPLATE_MIGRATION_SINGLE, data: item.data });
         break;
 
+      case JobName.STORAGE_TEMPLATE_MIGRATION_SINGLE:
+        await this.jobRepository.queue({ name: JobName.GENERATE_JPEG_THUMBNAIL, data: item.data });
```

Comment:
Manually running exif could trigger this migration job, which then triggers all the ML jobs. We don't want that to happen, unless it is the first time this job is running for an uploaded image.

Can you add a check `if(item.data.source === 'upload')` then run the thumbnail job? Otherwise do nothing.